modules/ROOT/pages/dataweave-cookbook-decode-encode.adoc

```diff
@@ -0,0 +1,160 @@
+= Decode and Encode Base64
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: studio, anypoint, transform, transformer, format, xml, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping
+
+The DataWeave examples show how to convert a file stream into Base64 and to convert a Base64 string into a file stream. They use a PDF file as the input and output.
+//LINK TO DW 1.0 EXAMPLES:
+include::partial$dataweave1-links.adoc[tag=dataweave1Examples]
+
+[[ex01]]
+== Encode a PDF File to Base64
+
+This example performs Base64 encoding on a PDF file. It simulates the transformation in the Read operation from the <<xml_config, XML configuration>> by converting a Base64 file stream (`application/octet-stream`) to Binary and outputting the Binary in the `text/plain` format. 
+
+The example uses the following functions:
+
+* `fromBase64` from the `dw::core::Binaries` module to transform a Base64 string to a binary value.
+*  `readUrl` from the `Core (dw::Core)` module to input a PDF file.
++
+The example assumes that the PDF file is in the `src/main/resources` directory of a Mule project in Anypoint Studio. It provides the name of the PDF file and specifies the MIME type as `"application/octet-stream"`. Using `"binary"` as the second `readUrl` argument is also valid for this example.
```

Comment:
```suggestion
The example assumes that the PDF file is in the `src/main/resources` directory of a Mule project in Anypoint Studio. It provides the name of the PDF file and specifies the MIME type as `application/octet-stream`. Using `binary` as the second `readUrl` argument is also valid for this example.
```

Reply:
the arguments are surrounded by quotation marks, so I needed to put quotes around them